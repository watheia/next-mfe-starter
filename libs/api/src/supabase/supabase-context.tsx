import { SupabaseClient } from '@supabase/supabase-js';
import { createContext, useContext } from 'react';

export type SupabaseContextType = {
  /**
   * Reference to supabase singleton.
   */
  supabase: SupabaseClient;
};

export const SupabaseContext = createContext<SupabaseContextType | null>(null);

export const useSupabase = (): SupabaseClient => {
  const ctx = useContext(SupabaseContext);
  if (ctx === null) {
    throw new Error(`useSupabase must be used within a SupabaseProvider.`);
  }

  return ctx.supabase;
};
