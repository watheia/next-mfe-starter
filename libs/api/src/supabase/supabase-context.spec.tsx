import { render } from '@testing-library/react';
import { SupaAuthExample } from './supabase-context.composition';

it('should render successfully', () => {
  const { container } = render(<SupaAuthExample />);
  expect(container).toBeInstanceOf(HTMLDivElement);
  expect(container).toMatchSnapshot();
});
