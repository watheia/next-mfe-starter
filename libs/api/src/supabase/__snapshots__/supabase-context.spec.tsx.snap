// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render successfully 1`] = `
<div>
  <div>
    rpc(fn, params, { head = false, count = null, } = {}) {
        const rest = this._initPostgRESTClient();
        return rest.rpc(fn, params, { head, count });
    }
  </div>
</div>
`;
